src/libraries/Native/native-binplace.proj

```diff
@@ -24,6 +24,10 @@
       <BinPlaceItem Condition="'$(BinPlaceTestRuntimePack)' != 'true'" Include="@(NativeBinPlaceItem)" />
       <FileWrites Include="@(NativeBinPlaceItem)" />
     </ItemGroup>
+    <ItemGroup Condition="'$(BinPlaceTestRuntimePack)' == 'true' and '$(TargetsMobile)' != 'true'">
+      <TestHostBinPlaceItem Include="%(NativeBinPlaceItem.Identity)" />
```

Comment:
Is this necessary? I think we can achieve the same by updating the condition to include into `BinPlaceItem`. Something like: `'$(BinPlaceTestRuntimePack)' != 'true' or '$(TargetsMobile)' != 'true'`?

Reply:
For my understanding ;-)

Can you give an example of what you think will work?

Reply:
Instead of adding this new item I think you can just update the above Item to:
```suggestion
      <BinPlaceItem Condition="'$(BinPlaceTestRuntimePack)' != 'true' or '$(TargetsMobile)' != 'true'" Include="@(NativeBinPlaceItem)" />
      <FileWrites Include="@(NativeBinPlaceItem)" />
    </ItemGroup>
```

BinPlaceItem should always contribute to the testhost directory. `BinPlaceTestRuntimePack` will only be true when `RuntimeFlavor == Mono`, if `TargetsMobile != true` that means we're building for Xamarin Desktop. Is my assumption right?

Reply:
Changing the code to
```
  <Target Name="GetBinPlaceItems">
    <ItemGroup>
      <NativeBinPlaceItem Include="$(NativeBinDir)*.dll" />
      <NativeBinPlaceItem Include="$(NativeBinDir)*.pdb" />
      <NativeBinPlaceItem Include="$(NativeBinDir)*.lib" />
      <NativeBinPlaceItem Include="$(NativeBinDir)*.a" />
      <NativeBinPlaceItem Include="$(NativeBinDir)*.bc" />
      <NativeBinPlaceItem Include="$(NativeBinDir)*.so" />
      <NativeBinPlaceItem Include="$(NativeBinDir)*.dbg" />
      <NativeBinPlaceItem Include="$(NativeBinDir)*.dylib" />
      <NativeBinPlaceItem Include="$(NativeBinDir)*.dwarf" />

      <BinPlaceItem Condition="'$(BinPlaceTestRuntimePack)' != 'true' or '$(TargetsMobile)' != 'true'" Include="@(NativeBinPlaceItem)" />
      <FileWrites Include="@(NativeBinPlaceItem)" />
    </ItemGroup>
  </Target>
```
The two testhost, before and after, are identical. However lib-runtime-packs seem to differ, where the one with this change has more files.
```
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Globalization.Native.a
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Globalization.Native.dylib
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Globalization.Native.dylib.dwarf
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.IO.Compression.Native.a
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.IO.Compression.Native.dylib
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.IO.Compression.Native.dylib.dwarf
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.IO.Ports.Native.a
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.IO.Ports.Native.dylib
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.IO.Ports.Native.dylib.dwarf
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Native.a
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Native.dylib
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Native.dylib.dwarf
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Net.Security.Native.a
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Net.Security.Native.dylib
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Net.Security.Native.dylib.dwarf
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Security.Cryptography.Native.Apple.a
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Security.Cryptography.Native.Apple.dylib
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Security.Cryptography.Native.Apple.dylib.dwarf
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Security.Cryptography.Native.OpenSsl.a
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Security.Cryptography.Native.OpenSsl.dylib
Only in lib-runtime-packs/runtimes/osx-x64/lib/netcoreapp5.0: libSystem.Security.Cryptography.Native.OpenSsl.dylib.dwarf
```

Reply:
I can run it again, but is this desired?

Reply:
@steveisok we want the Xamarin Desktop runtime packs to have the native assets produced from libraries, right?

Reply:
Actually that's wrong. Because it is going under `lib`. This is not desired. Thanks for checking @mdh1418 -- then we can't use my suggestion here 😄

Reply:
Yes, that is correct.  They should have libs and native just like mobile.  Testhost should still exist for desktop because we're not quite ready to go all the way yet.

Reply:
I answered previously assuming your question.  The desktop structure for `lib-runtime-packs` should mirror mobile.  Libs contain the assemblies and native contains the native assets (mono runtime + system.native + SPC).

Reply:
Right. My suggestion here unfortunately doesn't work 😄. Whenever we start producing self contained assets I guess we can revisit and see if there is a better way.